gcp-auth-extension/src/main/java/io/opentelemetry/contrib/gcp/auth/GcpAuthAutoConfigurationCustomizerProvider.java

```diff
@@ -95,20 +98,36 @@ private static SpanExporter addAuthorizationHeaders(
   }
 
   private static Map<String, String> getRequiredHeaderMap(GoogleCredentials credentials) {
-    Map<String, String> gcpHeaders = new HashMap<>();
+    Map<String, List<String>> gcpHeaders;
     try {
-      credentials.refreshIfExpired();
+      // this also refreshes the credentials, if required
+      gcpHeaders = credentials.getRequestMetadata();
     } catch (IOException e) {
       throw new GoogleAuthException(Reason.FAILED_ADC_REFRESH, e);
     }
-    gcpHeaders.put("Authorization", "Bearer " + credentials.getAccessToken().getTokenValue());
-    String configuredQuotaProjectId =
-        ConfigurableOption.GOOGLE_CLOUD_QUOTA_PROJECT.getConfiguredValueWithFallback(
-            credentials::getQuotaProjectId);
-    if (configuredQuotaProjectId != null && !configuredQuotaProjectId.isEmpty()) {
-      gcpHeaders.put(QUOTA_USER_PROJECT_HEADER, configuredQuotaProjectId);
+    // flatten list
+    Map<String, String> flattenedHeaders =
+        gcpHeaders.entrySet().stream()
+            .collect(
+                Collectors.toMap(
+                    Map.Entry::getKey,
+                    entry ->
+                        entry.getValue().stream()
+                            .filter(Objects::nonNull) // Filter nulls
+                            .filter(s -> !s.isEmpty()) // Filter empty strings
+                            .collect(Collectors.joining(",")),
+                    (v1, v2) -> v2 // Merge function - take the last seen value
```

Comment:
Does the overload without a merge function work? It seems like you would never get a key collision since you started with a map